src/pages/recurring-expenses/edit/Edit.tsx

```diff
@@ -0,0 +1,176 @@
+/**
+ * Invoice Ninja (https://invoiceninja.com).
+ *
+ * @link https://github.com/invoiceninja/invoiceninja source repository
+ *
+ * @copyright Copyright (c) 2022. Invoice Ninja LLC (https://invoiceninja.com)
+ *
+ * @license https://www.elastic.co/licensing/elastic-license
+ */
+
+import { RecurringExpenseStatus } from 'common/enums/recurring-expense-status';
+import { route } from 'common/helpers/route';
+import { useTitle } from 'common/hooks/useTitle';
+import { RecurringExpense } from 'common/interfaces/recurring-expenses';
+import { ValidationBag } from 'common/interfaces/validation-bag';
+import { useRecurringExpenseQuery } from '../common/queries';
+import { Page } from 'components/Breadcrumbs';
+import { Dropdown } from 'components/dropdown/Dropdown';
+import { DropdownElement } from 'components/dropdown/DropdownElement';
+import { Default } from 'components/layouts/Default';
+import { Tab, Tabs } from 'components/Tabs';
+import { AdditionalInfo } from '../create/components/AdditionalInfo';
+import { Details } from '../create/components/Details';
+import { Notes } from '../create/components/Notes';
+import { TaxSettings } from '../create/components/Taxes';
+import { useEffect, useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import { useParams } from 'react-router-dom';
+import { useBulk } from './hooks/useBulk';
+import { useSave } from './hooks/useSave';
+import { useToggleStartStop } from './hooks/useToggleStartStop';
+
+export function Edit() {
+  const { documentTitle } = useTitle('recurring_expense');
+  const { id } = useParams();
+  const { data } = useRecurringExpenseQuery({ id });
+
+  const [t] = useTranslation();
+  const toggleStartStop = useToggleStartStop();
+
+  const pages: Page[] = [
+    { name: t('recurring_expenses'), href: '/recurring_expenses' },
+    { name: t('edit_recurring_expense'), href: route('/recurring_expenses/:id', { id }) },
+  ];
+
+  const tabs: Tab[] = [
+    {
+      name: t('edit'),
+      href: route('/recurring_expenses/:id/edit', { id }),
+    },
+    {
+      name: t('documents'),
+      href: route('/recurring_expenses/:id/documents', { id }),
+    },
+  ];
+
+  const [expense, setExpense] = useState<RecurringExpense>();
```

Comment:
As previously noted, this should be part of Jotai atom. Also, please use "recurringExpense, setRecurringExpense" etc.

Reply:
Jotai atom used, and appropriate naming used too